src/hotspot/share/opto/vectornode.cpp

```diff
@@ -2085,6 +2085,56 @@ Node* VectorBlendNode::Identity(PhaseGVN* phase) {
   }
   return this;
 }
+static bool is_replicate_uint_constant(const Node* n) {
+  return n->Opcode() == Op_Replicate &&
+         n->in(1)->is_Con() &&
+         n->in(1)->bottom_type()->isa_long() &&
+         n->in(1)->bottom_type()->is_long()->get_con() <= 4294967295L;
+}
+
+static bool is_lower_doubleword_mask_pattern(const Node* n) {
+  return n->Opcode() == Op_AndV &&
+         (is_replicate_uint_constant(n->in(1)) ||
+          is_replicate_uint_constant(n->in(2)));
+}
+
+static bool is_clear_upper_doubleword_uright_shift_pattern(const Node* n) {
+  return n->Opcode() == Op_URShiftVL &&
+         n->in(2)->Opcode() == Op_RShiftCntV && n->in(2)->in(1)->is_Con() &&
+         n->in(2)->in(1)->bottom_type()->isa_int() &&
+         n->in(2)->in(1)->bottom_type()->is_int()->get_con() >= 32;
+}
+
+static bool has_vector_elements_fit_uint(Node* n) {
+  return is_lower_doubleword_mask_pattern(n) ||             // (AndV     SRC (Replicate C)) where C <= 0xFFFFFFFF
+         is_clear_upper_doubleword_uright_shift_pattern(n); // (URShiftV SRC S) where S >= 32
+}
+
+static bool has_vector_elements_fit_int(Node* n) {
+  auto is_cast_integer_to_long_pattern = [](const Node* n) {
```

Comment:
I like how you use lambda expressions for node predicates. 
Please, shape `has_vector_elements_fit_uint()` in a similar fashion.